packages/surveys/src/components/questions/ConsentQuestion.tsx

```diff
@@ -24,7 +28,34 @@ export default function ConsentQuestion({
   onBack,
   isFirstQuestion,
   isLastQuestion,
+  ttcObj,
+  setTtcObj,
 }: ConsentQuestionProps) {
+  const [startTime, setStartTime] = useState(performance.now());
```

Comment:
We have this code in all question types. Can we use a helper function or hook instead? e.g.

```ts
import { useEffect, useState } from 'react';

function useTtc(questionId, ttc, setTtc) {
  const [startTime, setStartTime] = useState(performance.now());

  useEffect(() => {
    setStartTime(performance.now());
  }, [questionId]);

  useEffect(() => {
    const handleVisibilityChange = () => {
      if (document.visibilityState === 'visible') {
        // Restart the timer when the tab becomes visible again
        setStartTime(performance.now());
      } else {
        const updatedTtc = getUpdatedTtc(
          ttc,
          questionId,
          performance.now() - startTime
        );
        setTtc(updatedTtc);
      }
    };

    // Attach the event listener
    document.addEventListener('visibilitychange', handleVisibilityChange);

    return () => {
      // Clean up the event listener when the component is unmounted
      document.removeEventListener('visibilitychange', handleVisibilityChange);
    };
  }, []);
}

export default useTtc;
```

```ts
import useTtc from '@/lib/useTtc';

function MyComponent({ question, ttcObj, setTtcObj }) {
  useVisibilityTimer(question.id, ttcObj, setTtcObj);

  // Rest of your component logic...
}
```